admin/notifiers/class-configuration-notifier.php

```diff
@@ -0,0 +1,175 @@
+<?php
+/**
+ * WPSEO plugin file.
+ *
+ * @package WPSEO\Admin\Notifiers
+ */
+
+/**
+ * Represents the logic for showing the notification.
+ */
+class WPSEO_Configuration_Notifier implements WPSEO_Listener {
+
+	/**
+	 * Option name used to determine whether the notice has been dismissed.
+	 *
+	 * @var string
+	 */
+	const META_NAME = 'wpseo-dismiss-configuration-notice';
+
+	/**
+	 * Default value.
+	 *
+	 * @var string
+	 */
+	const META_VALUE = 'yes';
+
+	/**
+	 * Should the notification be shown.
+	 *
+	 * @var bool
+	 */
+	protected $show_notification;
+
+	/**
+	 * Constructs the object by setting the show notification property based the given options.
+	 */
+	public function __construct() {
+		$this->show_notification = WPSEO_Options::get( 'show_onboarding_notice', false );
+	}
+
+	/**
+	 * Returns the content of the notification.
+	 *
+	 * @return string A string with the notification HTML, or empty string when no notification is needed.
+	 */
+	public function notify() {
+		if ( ! $this->show_notification() ) {
+			return $this->re_run_notification();
+		}
+
+		if ( WPSEO_Options::get( 'started_configuration_wizard' ) ) {
+			return $this->continue_notification();
+		}
+
+		return $this->first_time_notification();
```

Comment:
If we change the related methods to _not_ return the message string, then we should also remove these 3 returns and just call the methods. The description should be updated as well.